src/multi/scanner_thread.py

```diff
@@ -4,16 +4,26 @@
 def split_processing(ports, num_splits, scan, range_low, range_high):
     split_size = (range_high-range_low) // num_splits
     threads = []
-    for i in range(num_splits):
+    start  = 0
+    before = split_size * num_splits
+    for end in range(split_size, before, split_size):
         # determine the indices of the list this thread will handle
-        start = i * split_size
+        # We can just use this for loop
+#         start = i * split_size
         # special case on the last chunk to account for uneven splits
-        end = range_high if i+1 == num_splits else (i+1) * split_size
+    # No point of comparing each time if looking for end
+    # We can create end
+#         end = range_high if i+1 == num_splits else (i+1) * split_size
         # create the thread
         threads.append(
             threading.Thread(target=scan, args=(ports, start, end)))
+        start = end
         threads[-1].start()  # start the thread we just created
+    # The range - end will be helpful for remaining
+    threads.append(
+        threading.Thread(target=scan, args=(ports, end, range_end)))
```

Comment:
Just asking, what's the reference for `range_end`?

Reply:
the for loop goes to before i.e. (num_splits - 1)*split_size and the remaining will be (range_end - end) so the last thread will cover that remaining scans. For example we want to scan port 40 to 8000 so, we will have to split in 20
398--> split_size
20--> num_splits
now the for loop will go to 19*398 or 7562 leaving the rest i.e. 7562 to 8000 for this piece of code

Reply:
this comes up when 8000 to 40 with 68 splits to each thread the calculation will tell you 117 split_size so the end will be equal to 7839, leaving 8000 - 7839(range_end - end) or 161 ports which will be covered by this statement. In general when (range_end - range_start) mod  num_splits != 0